### PR TITLE
Chat message editor with mentions support

### PR DESCRIPTION
**Pull request by crui3er**
Chat message editor is implemented based on slate editor (react component)

image

### CODE REVIEW

**Review by alexis-kochetov — CHANGES_REQUESTED** [review]

**Review by alexyakunin — CHANGES_REQUESTED** [review]

File: src/dotnet/Chat.UI.Blazor/Components/ChatMessageEditor/ChatMessageEditor.razor [hunk 1]
```diff
@@ -7,74 +7,90 @@
...
+                    <ul class="message-attachments flex ml-1 mb-1 pt-4 pb-2 px-2 gap-5 overflow-x-auto">
+                        @foreach (var attachment in _attachments) {
+                            <li class="attachment-item p-1.5 bg-zinc-100 rounded-md inline-flex-y box-content">
+                                <div class="h-full flex-y relative">
+                                    <div class="mt-auto h-full flex justify-center relative min-h-0">
+                                        @if (attachment.IsImage) {
+                                            <img class="object-contain rounded-md max-w-full" src="@attachment.Url" alt="image"/>
+                                        } else {
+                                            <div class="h-40 flex">
+                                                <svg class="text-secondary fill-current h-20 w-20 self-center" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24">
```
[the remark this answers is not in the document]
Comment (crui3er) [reply src/dotnet/Chat.UI.Blazor/Components/ChatMessageEditor/ChatMessageEditor.razor:38]: I agree. I'd extract AttachmentsPanel in a separate control after this PR is merged.

File: src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionListView.razor [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+                 @onmouseover="@(() => OnMouseOver(iLocal))"
+                 @onclick="@(() => OnClick(item))">
+                @item.Name
+            </div>
+        }
+    </div>
+}
+
+@code {
+    [CascadingParameter]
```
[the remark this answers is not in the document]
Comment (crui3er) [reply src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionListView.razor:27]: done

File: src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionListContextProvider.razor [hunk 3]
```diff
@@ -0,0 +1,12 @@
+@namespace ActualChat.Chat.UI.Blazor.Components
```
[the remark this answers is not in the document]
Comment (crui3er) [reply src/dotnet/Chat.UI.Blazor/Components/MentionList/MentionListContextProvider.razor:1]: done

File: src/dotnet/Chat.Client/ClientDefs.cs [hunk 4]
```diff
@@ -58,6 +58,9 @@ public interface IChatsClientDef
     [Get(nameof(GetUserPeerChatId))]
     Task<string?> GetUserPeerChatId(Session session, string chatAuthorId, CancellationToken cancellationToken);
 
+    [Get(nameof(GetChatAuthors))]
+    Task<Author[]> GetChatAuthors(Session session, string chatId, CancellationToken cancellationToken);
```
[the remark this answers is not in the document]
Comment (crui3er) [reply src/dotnet/Chat.Client/ClientDefs.cs]: Fixed. Now I get directly MentionListItem's from the server.

File: src/dotnet/Chat.Contracts/IChats.cs [hunk 5]
```diff
@@ -55,6 +57,9 @@ public interface IChats
     [ComputeMethod(KeepAliveTime = 1)]
     Task<string?> GetUserPeerChatId(Session session, string chatAuthorId, CancellationToken cancellationToken);
 
+    [ComputeMethod(KeepAliveTime = 1)]
+    Task<Author[]> GetChatAuthors(Session session, string chatId, CancellationToken cancellationToken);
```
[the remark this answers is not in the document]
Comment (crui3er) [reply src/dotnet/Chat.Contracts/IChats.cs]: Fixed

**Pull request merged**